### PR TITLE
luaPackages: refactor custom busted nlua checkPhases; {busted, nlua}: add meta.mainProgram

### PR DESCRIPTION
**Pull request by khaneliman**
## Things done
While working on https://github.com/NixOS/nixpkgs/pull/352277 I enabled the checkPhase for vimPlugins. I noticed that nativeCheckInputs don't propogate down to vimPlugin's with the luaAttr. Refactoring the checkPhase to carry the store path of the `busted` and `nlua` binaries. 

Breaking this off to remove the noise from the neovimRequireCheckHook diff. 

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mrcjkb** [comment]
question: wouldn't it be better to propagate the `nativeCheckInputs` to the vim plugin, for example in the `neovimUtils.buildNeovimPlugin` function?
Asking, because we will probably want to fix luarocks-nix to add a luarocks package's `test_dependencies` to the `nativeCheckInputs`.

**Comment by khaneliman** [comment]
> question: wouldn't it be better to propagate the `nativeCheckInputs` to the vim plugin, for example in the `neovimUtils.buildNeovimPlugin` function? Asking, because we will probably want to fix luarocks-nix to add a luarocks package's `test_dependencies` to the `nativeCheckInputs`.

Yeah, ideally that would work. I just wasn't familiar with those enough, yet. I can take a look.

File: pkgs/development/lua-modules/overrides.nix [hunk 1]
```diff
@@ -534,22 +535,20 @@ in
 
   lz-n  = prev.lz-n.overrideAttrs(oa: {
     doCheck = lua.luaversion == "5.1";
-    nativeCheckInputs = [ final.nlua final.busted ];
     checkPhase = ''
       runHook preCheck
       export HOME=$(mktemp -d)
-      busted --lua=nlua
+      ${lib.getExe final.busted} --lua=${lib.getExe final.nlua}
```
Comment (teto) [inline pkgs/development/lua-modules/overrides.nix:541]: any reason to do so ? It makes the code more obscure and harder to copy/paste.

Comment (teto) [reply pkgs/development/lua-modules/overrides.nix:541]: ha because you removed nativeCheckInputs

**Comment by teto** [comment]
yeah it's better to propagate rather than hardcoding paths which is more painful to work with IMO. 
if you dont like nativeCheckInputs, we can rename it to checkInputs. Not sure why it doesn't propagate though, happy to merge such a fix.

**Comment by khaneliman** [comment]
Actually ended up being a bug in the migration to use checkPhase in the other PR... lol

**Pull request closed without merging**